test/unit/math/test_ad_matvar.hpp

```diff
@@ -398,27 +388,50 @@ template <typename... Types, typename F, typename... EigMats>
 void expect_ad_matvar_impl(const ad_tolerances& tols, const F& f,
                            const EigMats&... x) {
   using stan::is_var;
+  using stan::is_var_matrix;
+  using stan::is_eigen;
   using stan::plain_type_t;
   using stan::math::promote_scalar_t;
   using stan::math::var;
   using stan::math::var_value;
   using stan::math::test::type_name;
+  
+  if(!stan::math::disjunction<is_var_matrix<Types>...>::value) {
+    FAIL() << "expect_ad_matvar requires at least one varmat input!" << std::endl;
+  }
+
+  if(!stan::math::disjunction<is_var<scalar_type_t<Types>>...>::value) {
+    FAIL() << "expect_ad_matvar requires at least one autodiff input!" << std::endl;
+  }
+
+  auto A_mv_tuple = std::make_tuple(convert_to_matvar<Types>(x)...);
+  auto A_vm_tuple = std::make_tuple(convert_to_varmat<Types>(x)...);
+
+  bool any_varmat = stan::math::apply([](const auto&... args) {
+      return stan::math::disjunction<is_var_matrix<decltype(args)>...>::value ||
+      stan::math::disjunction<stan::math::conjunction<is_std_vector<decltype(args)>, is_var_matrix<value_type_t<decltype(args)>>>...>::value;
+    }, A_vm_tuple);
+
+  if(!any_varmat) {
+    SUCCEED(); // If no varmats are created, skip this test
+    return;
+  }
```

Comment:
When would this happen?

Reply:
If a function was signature `vector, real` and these tests tried to check the `matvar, varmat` version of the function (which it will), then the `matvar, matvar` and `matvar, varmat` both use the C++ types `Eigen::Matrix<var, -1, 1>, var` so there's no point testing it.

It's important to actually not test this too, cause the return type will be matvar if there are not varmat arguments.

Three different notations for argument types in one sentence! Nice!